src/ess/amor/load.py

```diff
@@ -33,43 +33,112 @@ def load_events(detector: RawDetector[Run]) -> RawEvents[Run]:
     return RawEvents[Run](data)
 
 
+class _Detector:
+    nBlades = sc.scalar(14)  # number of active blades in the detector
+    nWires = sc.scalar(32)  # number of wires per blade
+    nStripes = sc.scalar(64)  # number of stipes per blade
+    angle = sc.scalar(5.1, unit='degree').to(
+        unit='rad'
+    )  # deg  angle of incidence of the beam on the blades (def: 5.1)
+    dZ = sc.scalar(4.0, unit='mm') * sc.sin(
+        angle
+    )  # mm  height-distance of neighboring pixels on one blade
+    dX = sc.scalar(4.0, unit='mm') * sc.cos(
+        angle
+    )  # mm  depth-distance of neighboring pixels on one blace
+    bladeZ = sc.scalar(10.455, unit='mm')  # mm  distance between detector blades
+    zero = 0.5 * nBlades.value * bladeZ  # mm  vertical center of the detector
+    distance = sc.scalar(
+        4000, unit='mm'
+    )  # mm  distance from focal point to leading blade edge
+
+
+def _pixel_coordinate_in_detector_system(pixelID):
+    """determine spatial coordinates and angles from pixel number"""
+    pixelID.unit = ''
+    (bladeNr, bPixel) = pixelID // (_Detector.nWires * _Detector.nStripes), pixelID % (
+        _Detector.nWires * _Detector.nStripes
+    )
+    (bZi, detYi) = (
+        bPixel // _Detector.nStripes,
+        bPixel % _Detector.nStripes,
+    )  # z index on blade, y index on detector
+    detZi = bladeNr * _Detector.nWires + bZi  # z index on detector
+    detX = bZi * _Detector.dX  # x position in detector
+
+    bladeAngle = (2.0 * sc.asin(0.5 * _Detector.bladeZ / _Detector.distance)).to(
+        unit='degree'
+    )
+    delta = (_Detector.nBlades / 2.0 - bladeNr) * bladeAngle - (
+        sc.atan(bZi * _Detector.dZ / (_Detector.distance + bZi * _Detector.dX))
+    ).to(unit='degree')
+
+    # z is in the direction of the center of the beam, y is the direction 'up'
+    pixelID.unit = None
+    return detYi, detZi, detX, delta
+
+
+def _pixel_coordinate_in_lab_frame(pixelID, nu):
+    _, detZi, detX, delta = _pixel_coordinate_in_detector_system(pixelID)
+
+    angle_to_horizon = (nu + delta).to(unit='rad')
+    distance_to_pixel = detX + _Detector.distance
+    # TODO: put the correct value here
+    global_X = sc.zeros(dims=pixelID.dims, shape=pixelID.shape, unit='mm')
+    global_Y = distance_to_pixel * sc.sin(angle_to_horizon)
+    global_Z = distance_to_pixel * sc.cos(angle_to_horizon)
+    return global_X, global_Y, global_Z
+
+
+def add_position(
+    detector: RawDetector[Run], mu: SampleRotation[Run], nu: DetectorRotation[Run]
+) -> RawEvents[Run]:
+    events = load_events(detector)
+    if 'event_time_zero' in events.coords:
+        events.bins.coords['event_time_zero'] = sc.bins_like(
+            events, fill_value=events.coords['event_time_zero']
+        )
+    events = events.bins.concat('event_time_zero').data.value
+    events = events.copy()
+    events.coords['detector_number'] = events.coords.pop('event_id')
```

Comment:
I don't know if this is meant to be temporary, but converting the `NXdetector` back to something like `NXevent_data` is obviously quite wasteful. Either we should work with the grouped-by-pixel data, or directly load `NXevent_data`.

Reply:
Yes it's temporary. The idea is to convert the workflow to work with the pixel binned data like in the other workflows (?), this was somthing @nvaytet also mentioned.
I don't think that will make us have to change too much of the rest of the workflow, but I will need some help to do it.

Reply:
So what was the reason for undoing the pixel-binning? Was it to get something to work quickly? Do you intend to remove this before finalizing the PR?